hivemind/moe/client/moe.py

```diff
@@ -375,19 +379,26 @@ def _collect_responses(
         timeout_total = float("inf") if timeout_total is None else timeout_total
         timeout_after_k_min = float("inf") if timeout_after_k_min is None else timeout_after_k_min
         num_successful_tasks = [0 for _ in range(num_samples)]
-        pending_samples = num_samples  # samples for which we have less than k_min results
+
+        samples_with_tasks = {sample_idx for sample_idx, _ in task_to_indices.values()}
+        pending_samples = len(samples_with_tasks)  # samples for which we have less than k_min results
+        assert pending_samples <= num_samples
```

Comment:
It's better to recompute the number of pending_samples, as some samples might have 0 tasks (for example, in `test_call_many`)